### PR TITLE
feat: support multi ingestion keys

### PR DESCRIPTION
**Pull request by YounixM**
Fixes:  #4213 

Screenshot 2024-06-03 at 4 35 28 PM

Screenshot 2024-06-03 at 4 35 42 PM

Screenshot 2024-06-03 at 4 35 50 PM

Screenshot 2024-06-03 at 4 35 59 PM

Screenshot 2024-06-03 at 4 36 07 PM

Screenshot 2024-06-03 at 4 36 16 PM

Screenshot 2024-06-03 at 4 36 25 PM

Screenshot 2024-06-03 at 4 36 39 PM

Screenshot 2024-06-03 at 4 36 46 PM

Screenshot 2024-06-03 at 4 36 53 PM

Screenshot 2024-06-03 at 4 37 08 PM

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (SagarRajput-7) [inline yarn.lock:1]: Please, remove this file

File: frontend/src/utils/app.ts [hunk 2]
```diff
@@ -15,6 +15,8 @@ export function extractDomain(email: string): string {
 export const isCloudUser = (): boolean => {
 	const { hostname } = window.location;
 
+	return true;
```
Comment (SagarRajput-7) [inline frontend/src/utils/app.ts:18]: Remove this ^

File: frontend/src/components/Tags/Tags.styles.scss [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		flex-wrap: wrap;
+	}
+
+	.ant-form-item {
+		margin-bottom: 0;
+	}
+
+	.ant-tag {
+		margin-right: 0;
+		background: white;
```
Comment (vikrantgupta25) [inline frontend/src/components/Tags/Tags.styles.scss:20]: css variables

File: frontend/package.json [hunk 4]
```diff
@@ -88,6 +88,7 @@
 		"lucide-react": "0.379.0",
 		"mini-css-extract-plugin": "2.4.5",
 		"papaparse": "5.4.1",
+		"rc-tween-one": "3.0.6",
```
Comment (vikrantgupta25) [inline frontend/package.json:91]: the last commit to this library was 2 years ago... doesn't seem like it's actively maintained

File: frontend/src/api/IngestionKeys/createIngestionKey.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+		});
+
+		return {
+			statusCode: 200,
+			error: null,
+			message: response.data.status,
+			payload: response.data.data,
+		};
+	} catch (error) {
+		return ErrorResponseHandler(error as AxiosError);
```
Comment (vikrantgupta25) [inline frontend/src/api/IngestionKeys/createIngestionKey.ts:25]: the errors won't be exposed to the consumer here.. the error handling for them won't work. please ensure if this is the expected case.

File: frontend/src/container/IngestionSettings/MultiIngestionSettings.tsx [hunk 6]
```diff
@@ -0,0 +1,1133 @@
...
+					<Form.Item name="tags" label="Tags">
+						<Tags tags={updatedTags} setTags={setUpdatedTags} />
+					</Form.Item>
+				</Form>
+			</Modal>
+		</div>
+	);
+}
+
+export default MultiIngestionSettings;
```
Comment (SagarRajput-7) [inline frontend/src/container/IngestionSettings/MultiIngestionSettings.tsx:1133]: NIT: We can divide this file into multiple, especially around the render return we have too many nested divs, so we can move those to multiple components if possible for better readability.

**Review by SagarRajput-7 — APPROVED** [review]

File: frontend/src/api/IngestionKeys/createIngestionKey.ts (hunk 5 above)
Comment (YounixM) [reply frontend/src/api/IngestionKeys/createIngestionKey.ts:25]: Yes. This works as expected.

File: frontend/package.json (hunk 4 above)
Comment (YounixM) [reply frontend/package.json:91]: It's fine. It's as per antd documentation. Working as expected.

File: frontend/src/container/IngestionSettings/MultiIngestionSettings.tsx (hunk 6 above)
Comment (YounixM) [reply frontend/src/container/IngestionSettings/MultiIngestionSettings.tsx:1142]: Yes. Will refactor this later

File: frontend/package.json (hunk 4 above)
Comment (YounixM) [reply frontend/package.json:91]: https://ant.design/components/tag#tag-demo-animation

File: yarn.lock (hunk 1 above)
Comment (YounixM) [reply yarn.lock:1]: ??

Comment (vikrantgupta25) [reply yarn.lock:1]: this is at the root of the repo.

**Review by SagarRajput-7 — APPROVED** [review]

**Pull request merged**